src/components/DayPickerRangeController.jsx

```diff
@@ -160,6 +160,8 @@ export default class DayPickerRangeController extends React.Component {
     if (focusedInput !== END_DATE) return false;
 
     if (startDate) {
+      if (minimumNights === 1) return isSameDay(day, startDate);
```

Comment:
I was trying a similar approach just now, but figured I'd describe it here first. I was worried about `clone()`'s performance, so I was doing `dayDiff = day.day() - startDate.day()`. Does this look better to you?